src/material-luxon-adapter/adapter/luxon-date-adapter.ts

```diff
@@ -0,0 +1,246 @@
+
+/** Configurable options for {@see LuxonDateAdapter}. */
+export interface MatLuxonDateAdapterOptions {
+  /**
+   * Turns the use of utc dates on or off.
+   * Changing this will change how Angular Material components like DatePicker output dates.
+   * {@default false}
+   */
+  useUtc: boolean;
+}
+
+/** InjectionToken for LuxonDateAdapter to configure options. */
+export const MAT_LUXON_DATE_ADAPTER_OPTIONS = new InjectionToken<MatLuxonDateAdapterOptions>(
+  'MAT_LUXON_DATE_ADAPTER_OPTIONS', {
+    providedIn: 'root',
+    factory: MAT_LUXON_DATE_ADAPTER_OPTIONS_FACTORY
+});
+
+
+/** @docs-private */
+export function MAT_LUXON_DATE_ADAPTER_OPTIONS_FACTORY(): MatLuxonDateAdapterOptions {
+  return {
+    useUtc: false
+  };
+}
+
+/** The default date names to use if Intl API is not available. */
+const DEFAULT_DATE_NAMES = range(31, i => String(i + 1));
+
+/** Creates an array and fills it with values. */
+function range<T>(length: number, valueFunction: (index: number) => T): T[] {
+  const valuesArray = Array(length);
+  for (let i = 0; i < length; i++) {
+    valuesArray[i] = valueFunction(i);
+  }
+  return valuesArray;
+}
+
+/** Adapts Luxon Dates for use with Angular Material. */
+@Injectable()
+export class LuxonDateAdapter extends DateAdapter<DateTime> {
+  private _useUTC: boolean;
+
+  constructor(@Optional() @Inject(MAT_DATE_LOCALE) dateLocale: string,
+    @Optional() @Inject(MAT_LUXON_DATE_ADAPTER_OPTIONS)
+    options?: MatLuxonDateAdapterOptions) {
+
+    super();
+    this._useUTC = options ? !!options.useUtc : false;
+    this.setLocale(dateLocale || DateTime.local().locale);
+  }
+
+  setLocale(locale: string) {
+    super.setLocale(locale);
+  }
+
+  getYear(date: DateTime): number {
+    return date.year;
+  }
+
+  getMonth(date: DateTime): number {
+    // Luxon works with 1-indexed months whereas our code expects 0-indexed.
+    return date.month - 1;
+  }
+
+  getDate(date: DateTime): number {
+    return date.day;
+  }
+
+  getDayOfWeek(date: DateTime): number {
+    return date.weekday;
+  }
+
+  getMonthNames(style: 'long' | 'short' | 'narrow'): string[] {
+    return Info.months(style, {locale: this.locale});
+  }
+
+  getDateNames(): string[] {
+    if (Info.features().intl) {
+      // At the time of writing, Luxon doesn't offer similar
+      // functionality so we have to fall back to the Intl API.
+      const dtf = new Intl.DateTimeFormat(this.locale, {day: 'numeric', timeZone: 'utc'});
+
+      return range(31, i => {
+        // Format a UTC date in order to avoid DST issues.
+        const date = DateTime.utc(2017, 1, i + 1).toJSDate();
+
+        // Strip the directionality characters from the formatted date.
+        return dtf.format(date).replace(/[\u200e\u200f]/g, '');
+      });
+    }
+    return DEFAULT_DATE_NAMES;
+  }
+
+  getDayOfWeekNames(style: 'long' | 'short' | 'narrow'): string[] {
+    return Info.weekdays(style, {locale: this.locale});
```

Comment:
Not sure if should be changed here or somewhere else, but the way this is, the weekdays on the month view of the calendar don't line up correctly with the dates because it starts on Monday instead of Sunday.  For example, April 1, 2019 was on a Monday, but this adapter causes the 'T' for Tuesday to appear above April 1 instead of the 'M'.